burn-tensor/src/tensor/api/check.rs

```diff
@@ -99,7 +99,7 @@ impl TensorCheck {
         check
     }
 
-    pub(crate) fn reshape_infer<const D: usize>(target: &[i64; D]) -> Self {
+    pub(crate) fn reshape_args<const D: usize>(target: &[i32; D]) -> Self {
```

Comment:
I think I would rename to `reshape_args_i32` to highlight the fact that the check is for one variant of reshape arguments.